TestContainerAddResources fails

```
--- FAIL: TestContainerAddResources (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0xc87264]

goroutine 132 [running]:
testing.tRunner.func1(0xc42020e960)
	/usr/local/go/src/testing/testing.go:711 +0x5d9
panic(0xe526a0, 0x15a2340)
	/usr/local/go/src/runtime/panic.go:491 +0x2a2
github.com/kata-containers/runtime/virtcontainers.(*Container).addResources(0xc42047de58, 0x118, 0x0)
	github.com/kata-containers/runtime/virtcontainers/_test/_obj_test/container.go:935 +0x394
github.com/kata-containers/runtime/virtcontainers.TestContainerAddResources(0xc42020e960)
	/home/jenkins/jenkins_slave/workspace/kata-containers-runtime-ubuntu-16-04-PR/go/src/github.com/kata-containers/runtime/virtcontainers/container_test.go:311 +0x5a3
testing.tRunner(0xc42020e960, 0xf4e4c0)
	/usr/local/go/src/testing/testing.go:746 +0x16d
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:789 +0x569
exit status 2
```

--- a/virtcontainers/api_test.go
+++ b/virtcontainers/api_test.go
@@ -1726,7 +1726,7 @@ func TestEnterContainerFailingContNotStarted(t *testing.T) {
 	cmd := newBasicTestCmd()
 
 	_, c, _, err = EnterContainer(p.ID(), contID, cmd)
-	if c != nil || err == nil {
+	if c == nil || err != nil {
 		t.Fatal()
 	}
 }

--- a/virtcontainers/container.go
+++ b/virtcontainers/container.go
@@ -432,6 +432,19 @@ func (c *Container) rollbackFailingContainerCreation() {
 	}
 }
 
+func (c *Container) checkBlockDeviceSupport() bool {
+	if !c.pod.config.HypervisorConfig.DisableBlockDeviceUse {
+		agentCaps := c.pod.agent.capabilities()
+		hypervisorCaps := c.pod.hypervisor.capabilities()
+
+		if agentCaps.isBlockDeviceSupported() && hypervisorCaps.isBlockDeviceHotplugSupported() {
+			return true
+		}
+	}
+
+	return false
+}
+
 // createContainer creates and start a container inside a Pod. It has to be
 // called only when a new container, not known by the pod, has to be created.
 func createContainer(pod *Pod, contConfig ContainerConfig) (c *Container, err error) {
@@ -456,8 +469,10 @@ func createContainer(pod *Pod, contConfig ContainerConfig) (c *Container, err er
 		}
 	}()
 
-	if err = c.hotplugDrive(); err != nil {
-		return
+	if c.checkBlockDeviceSupport() {
+		if err = c.hotplugDrive(); err != nil {
+			return
+		}
 	}
 
 	// Attach devices
@@ -683,15 +698,6 @@ func (c *Container) processList(options ProcessListOptions) (ProcessList, error)
 }
 
 func (c *Container) hotplugDrive() error {
-	agentCaps := c.pod.agent.capabilities()
-	hypervisorCaps := c.pod.hypervisor.capabilities()
-
-	if c.pod.config.HypervisorConfig.DisableBlockDeviceUse ||
-		!agentCaps.isBlockDeviceSupported() ||
-		!hypervisorCaps.isBlockDeviceHotplugSupported() {
-		return nil
-	}
-
 	dev, err := getDeviceForPath(c.rootFs)
 
 	if err == errMountPointNotFound {

--- a/virtcontainers/container_test.go
+++ b/virtcontainers/container_test.go
@@ -210,6 +210,12 @@ func TestContainerAddDriveDir(t *testing.T) {
 		id:         testPodID,
 		storage:    fs,
 		hypervisor: &mockHypervisor{},
+		agent:      &noopAgent{},
+		config: &PodConfig{
+			HypervisorConfig: HypervisorConfig{
+				DisableBlockDeviceUse: false,
+			},
+		},
 	}
 
 	contID := "100"
@@ -258,7 +264,6 @@ func TestContainerAddDriveDir(t *testing.T) {
 	if container.state.Fstype == "" || !container.state.HotpluggedDrive {
 		t.Fatal()
 	}
-
 }
 
 func TestCheckPodRunningEmptyCmdFailure(t *testing.T) {
@@ -307,7 +312,10 @@ func TestContainerAddResources(t *testing.T) {
 		CPUQuota:  5000,
 		CPUPeriod: 1000,
 	}
-	c.pod = &Pod{hypervisor: &mockHypervisor{}}
+	c.pod = &Pod{
+		hypervisor: &mockHypervisor{},
+		agent:      &noopAgent{},
+	}
 	err = c.addResources()
 	assert.Nil(err)
 }

--- a/virtcontainers/pkg/vcmock/container.go
+++ b/virtcontainers/pkg/vcmock/container.go
@@ -30,6 +30,10 @@ func (c *Container) Pod() vc.VCPod {
 
 // Process implements the VCContainer function of the same name.
 func (c *Container) Process() vc.Process {
+	// always return a mockprocess with a non-zero Pid
+	if c.MockProcess.Pid == 0 {
+		c.MockProcess.Pid = 1000
+	}
 	return c.MockProcess
 }
 

